srtcore/core.cpp

```diff
@@ -5027,6 +5037,7 @@ void CUDT::acceptAndRespond(const sockaddr_any& peer, const CPacket& hspkt, CHan
         throw CUDTException(MJ_SETUP, MN_REJECTED, 0);
     }
 
+    updateAfterSrtHandshake(w_hs.m_iVersion);
```

Comment:
This is so because the `m_ConnRes` in this case is not deployed with the incoming handshake data, which is written into this `w_hs`, which is actually an input-output parameter here (it's later written to by `createSrtHandshake`). In `postConnect`, which is called in case when agent is an initiator (caller), this data is deployed into `m_ConnRes` and therefore it can go there. Not sure why this difference or whether it should be made to `m_ConnRes` also in listener case - this may bear further investigation. At least for tests I made with checking how this is interpreted, TsbPd settings are set correctly and the same as in the current master version.